pkg/cmd/pr/view/view.go

```diff
@@ -84,21 +89,39 @@ func viewRun(opts *ViewOptions) error {
 	}
 	apiClient := api.NewClientFromHTTP(httpClient)
 
-	pr, _, err := shared.PRFromArgs(apiClient, opts.BaseRepo, opts.Branch, opts.Remotes, opts.SelectorArg)
+	pr, repo, err := shared.PRFromArgs(apiClient, opts.BaseRepo, opts.Branch, opts.Remotes, opts.SelectorArg)
 	if err != nil {
 		return err
 	}
 
-	openURL := pr.URL
 	connectedToTerminal := opts.IO.IsStdoutTTY() && opts.IO.IsStderrTTY()
 
 	if opts.BrowserMode {
+		openURL := pr.URL
 		if connectedToTerminal {
 			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
 		}
 		return utils.OpenInBrowser(openURL)
 	}
 
+	if opts.Comments {
+		var s *spinner.Spinner
+		if connectedToTerminal {
+			s = utils.Spinner(opts.IO.ErrOut)
+			utils.StartSpinner(s)
```

Comment:
The IO object has methods to start/stop the progress indicator, which you may call unconditionally! It will make this section simpler.